index.bs

```diff
@@ -3406,6 +3421,68 @@ and |session|:
 
 </div>
 
+<div algorithm="collect nodes using accessibility attributes">
+To <dfn>collect nodes using accessibility attributes</dfn> with given |context nodes|, |selector|,
+|maximum returned node count|, and |returned nodes|:
+
+1. If |returned nodes| is null:
+
+  1. Set |returned nodes| to an empty [=/list=].
+
+1. For each |context node| in |context nodes|:
+
+  1. Let |match| be true.
+
+  1. If |selector|'s <code>role</code> is present:
+
+    1. Let |role| be the [=computed role=] of |context node|.
+
+    1. If |selector|'s <code>role</code> [=is|is not=] |role|:
+
+      1. Set |match| to false.
+
+  1. If |selector|'s <code>name</code> is present:
+
+    1. Let |name| be the [=accessible name=] of |context node|.
+
+    1. If |selector|'s <code>name</code> [=is|is not=] |name|:
+
+      1. Set |match| to false.
+
+  1. If |match| is true:
+
+    1. If |maximum returned node count| is not null and [=list/size=] of |returned
+      nodes| is greater or equal to |maximum returned node count|:
+
+      1. Return [=success=] with data |returned nodes|.
```

Comment:
I mean that it shouldn't say "is greater [than] or equal to" if it's an invariant that it can't be greater than, which the later assert requires that it is. So I think we should remove "if greater or".

Reply:
Also, I think a break would continue the recursion so it has to be a return to prevent redundant iterations?

Reply:
Unless I'm misunderstanding the semantics there's only one loop here? And neither break nor return will propogate through recursive calls. Anyway either is fine.

Reply:
Yeah, I have misread the indentation